### PR TITLE
SALTO-5456: hide workflow transitionId and add it in the deployment

### PR DESCRIPTION
**Pull request by shayc331**
_hide workflow transitionId and add it in the deployment_

---

_Additional context for reviewer_
in this pr: 
* add transition_ids filter
* fix workflow schema bug
* fix workflow_filter test

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/jira-adapter/test/filters/workflowV2/workflow_filter.test.ts [hunk 1]
```diff
@@ -61,6 +61,7 @@ describe('jiraWorkflowFilter', () => {
   const TRANSITION_NAME_TO_KEY:Record<string, string> = {
     Create: naclCase('Create::From: none::Initial'),
     Done: naclCase('Done::From: any status::Global'),
+    ToStatus2: naclCase('ToStatus2::From: Create::Directed'),
```
Comment (shayc331) [inline packages/jira-adapter/test/filters/workflowV2/workflow_filter.test.ts:64]: note: 
this test changes are leftover from this [task](https://salto-io.atlassian.net/browse/SALTO-5362)
(the transition list transformed into a map)

File: packages/jira-adapter/src/filters/workflowV2/types.ts [hunk 2]
```diff
@@ -72,9 +72,9 @@ type WorkflowScope = {
 
 export type Workflow = {
   name: string
-  version: WorkflowVersion
+  version?: WorkflowVersion
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/workflowV2/types.ts:75]: note: 
it was a bug, in addition there are no version nor id

File: packages/jira-adapter/src/filters/workflowV2/transition_ids.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  name: 'transitionIdsFilter',
+  preDeploy: async changes => {
+    changes
+      .filter(isInstanceChange)
+      .filter(isAdditionOrModificationChange)
+      .map(getChangeData)
+      .filter(isWorkflowV2Instance)
+      .forEach(addTransitionIds)
+  },
+})
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/workflowV2/transition_ids.ts:47]: Are the IDs hidden after deployment?

File: packages/jira-adapter/test/filters/workflowV2/transition_ids.test.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    })
+
+    it('should not fail if transitionId is invalid', async () => {
+      instance.value.transitions.transition2.id = 'invalid'
+      await filter.preDeploy([toChange({ after: instance })])
+      expect(instance.value.transitions.transition1.id).toBe('1')
+      expect(instance.value.transitions.transition2.id).toBe('invalid')
+      expect(instance.value.transitions.transition3.id).toBe('2')
+    })
+  })
```
Comment (IdoZakk) [inline packages/jira-adapter/test/filters/workflowV2/transition_ids.test.ts:64]: Add   a test to make sure you do not take the first number and also that you handle holes correctly (can be done in one...)

**Review by IdoZakk — APPROVED** [review]
💯 
I am approving (look at my comments), but we will still need to talk to Neta

File: packages/jira-adapter/src/filters/workflowV2/transition_ids.ts (hunk 3 above)
Comment (shayc331) [reply packages/jira-adapter/src/filters/workflowV2/transition_ids.ts:47]: yes

File: packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts [hunk 5]
```diff
@@ -181,7 +181,11 @@ const createWorkflowInstances = async ({
         response.data.workflows.map(async workflow => {
           convertTransitionParametersFields(workflow.transitions, convertParametersFieldsToList)
           convertPropertiesToList([...(workflow.statuses ?? []), ...(workflow.transitions ?? [])])
-          // convert transition list to map
```
[the remark this answers is not in the document]
Comment (shayc331) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:184]: yes thank you

**Pull request merged**